### PR TITLE
solving #11

### LINKED ISSUE #11: Full typing for each function
Some functions throughout our code are not fully typed. Need to go in and ensure all return types and arguments are properly typed

### PR DESCRIPTION
**Pull request by Siddhesh-Agarwal**
trying to close #11 
- added typing
- created a new BaseClass for type of parameter

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,160 @@
...
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# Distribution / packaging
+.Python
```
Comment (ishandhanani) [inline .gitignore:10]: @Siddhesh-Agarwal - not sure if we need all of these in the gitignore at the moment based on our current codebase

File: src/inference/inference.py [hunk 2]
```diff
@@ -1,37 +1,48 @@
-import torch
 import os
-from peft import AutoPeftModelForCausalLM
-from transformers import AutoTokenizer, pipeline
-import gradio as gr
+from typing import List
+
+import gradio as gr  # type: ignore
```
Comment (ishandhanani) [inline src/inference/inference.py:4]: I see this `type: ignore` comment in a couple different places. Could you explain the reasoning @Siddhesh-Agarwal

**Comment by ishandhanani** [comment]
@Siddhesh-Agarwal - thank you so much for the PR! A couple more questions. 

1. Did you format with `black`?
2. Can you walk through the reasoning for the BaseClass?

**Comment by Siddhesh-Agarwal** [comment]
1. Yes I formatted using isort followed by black.
2. Thanks for asking this, I used this BaseClass as a temporary solution because I was unable to figure out the data type of the parameter. However, upon thoroughly reading the code, I found out the attributes of the class. So I temporarily created this base class to solve that problem. Perhaps you could help me with the datatype of the parameter?

File: src/inference/inference.py (hunk 2 above)
Comment (Siddhesh-Agarwal) [reply src/inference/inference.py:4]: This is used by `mypy` to ignore the type of the variable (sometimes due to incompatible datatypes). I'll remove it if it is unnecessary for this project.

File: .gitignore (hunk 1 above)
Comment (Siddhesh-Agarwal) [reply .gitignore:10]: Well, I guess we can trim it down to your original `.gitignore` + virtual environments.

Comment (ishandhanani) [reply .gitignore:10]: Yea that would be great if you could add a commit to fix that

File: src/inference/inference.py (hunk 2 above)
Comment (ishandhanani) [reply src/inference/inference.py:4]: I don't believe it is needed for this

**Comment by ishandhanani** [comment]
I can help with that @Siddhesh-Agarwal. Can you link which parameters you're confused about in a comment and I can walk through them

**Pull request closed without merging**